### PR TITLE
feat: v1.1 graffiti support

### LINKED ISSUE #148: Add graffiti to UniRep v1.1
We can add Graffiti values to UniRep v1.1 by storing a timestamp with the graffiti. Then in the circuit we take the graffiti value with the largest timestamp value.

### PR DESCRIPTION
**Pull request by vimwitch**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Closes #148 

## What is the new behavior?

Adds `graffiti` and `timestamp` fields to attestations. Changes the prove reputation circuit to only prove the rep in the state tree, not the epoch tree.

Removes the public signals from the `aggregateEpochKeys` proof.

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by vivianjeng — COMMENTED** [review]
I think the functions in `Synchronizer` and `UserState` should be covered in tests
e.g. check if `Synchronizer` saves event correctly, trees are correctly generated, `UserState` can generate correct proofs...

File: packages/contracts/test/userStateTransition.test.ts [hunk 1]
```diff
@@ -572,7 +624,7 @@ describe('User State Transition', function () {
                 .userStateTransition(publicSignals, proof)
             await tx.wait()
             await expect(tx)
-                .to.emit(unirepContract, 'NewGSTLeaf')
+                .to.emit(unirepContract, 'StateTreeLeaf')
                 .withArgs(x + 1, accounts[1].address, 0, publicSignals[1])
             await expect(tx)
                 .to.emit(unirepContract, 'UserStateTransitioned')
```
[the remark this answers is not in the document]
Comment (vimwitch) [reply packages/contracts/test/userStateTransition.test.ts:630]: This is tested in the core package. But we need more tests for everything

File: packages/contracts/scripts/deploy.mjs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const epochLength = 12 * 60 * 60 // seconds
+const maxReputationBudget = 10
+
+const [signer] = await ethers.getSigners()
+const unirep = await deployUnirep(signer, {
+    attestingFee,
+    numEpochKeyNoncePerEpoch,
+    stateTreeDepth: STATE_TREE_DEPTH,
+    epochTreeDepth: EPOCH_TREE_DEPTH,
+})
```
Comment (vivianjeng) [inline packages/contracts/scripts/deploy.mjs:16]: now the settings are
```ts
{
        stateTreeDepth?: BigNumberish
        epochTreeDepth?: BigNumberish
        numEpochKeyNoncePerEpoch?: BigNumberish
        emptyEpochTreeRoot?: BigNumberish
    }
```

**Review by vivianjeng — CHANGES_REQUESTED** [review]
There is still a `globalStateTree` function
https://github.com/Unirep/Unirep/blob/3273d920d419cc82e84c54f3181b325418123fde/packages/core/src/Synchronizer.ts#L30

and there are still some `GST` are not changes in this PR

**Comment by vimwitch** [comment]
Ok, should be fixed

**Comment by vivianjeng** [comment]
Search for `GST` there are still many variables remained
https://github.com/Unirep/Unirep/blob/280b443208e86c75322fedda01ec5cb3bb1df7b1/packages/core/src/UserState.ts#L173

https://github.com/Unirep/Unirep/blob/3273d920d419cc82e84c54f3181b325418123fde/packages/core/src/Synchronizer.ts#L32

https://github.com/Unirep/Unirep/blob/280b443208e86c75322fedda01ec5cb3bb1df7b1/packages/circuits/test/utils.ts#L287

Can you check them in your local?
Since they are not in the pr

**Comment by vimwitch** [comment]
Ok all instances should be fixed

**Comment by vivianjeng** [comment]
https://github.com/Unirep/Unirep/blob/d1f3c4b87f6dff0b6301bf38ba82de60d808c497/packages/circuits/test/utils.ts#L351

https://github.com/Unirep/Unirep/blob/d1f3c4b87f6dff0b6301bf38ba82de60d808c497/packages/core/src/Synchronizer.ts#L81

some comments still remain.
we don't need the type of `circuitInput`?
I think we can maintain it so in `@unirep/circuits` and `@unirep/core` we can more easily create circuitInputs and don't need to look at circom files

**Comment by vimwitch** [comment]
We don't use the ts definitions in the prover implementations. I think we can re-add them later, but for now we can use the [docs](https://developer.unirep.io/docs/circuits-api/circuits#user-state-transition-proof).

> some comments still remain.

I think it's okay for now, we can remove them as we go.

**Review by vivianjeng — APPROVED** [review]

**Pull request merged**